src/index.js

```diff
@@ -269,14 +269,6 @@ const extendCodeBlockWithSyntaxHighlighting = (settings) => {
 			);
 		},
 
-		save({ attributes }) {
-			return (
-				<pre>
-					<code>{attributes.content}</code>
-				</pre>
-			);
-		},
-
```

Comment:
By deleting this here, we can just reuse what the overridden block is [already doing](https://github.com/WordPress/gutenberg/blob/050f0333e5e4dbbe00b8944fbb61c72992a4e6e9/packages/block-library/src/code/save.js#L6-L12), but with `escape()`:

```js
import { escape } from './utils';

export default function save( { attributes } ) {
	return (
		<pre>
			<code>{ escape( attributes.content ) }</code>
		</pre>
	);
}
```

Reply:
Actually, removing this is causing problems. In WordPress 5.4.1 without Gutenberg active, I'm getting an error where the `save` function in the `blockType` is undefined:

![image](https://user-images.githubusercontent.com/134745/85251017-0f26d880-b40d-11ea-9196-6db656b033f2.png)

Reply:
For some reason `settings.save` is defined up front, after being filtered it gets dropped, even though all of `settings` is being returned:

https://github.com/westonruter/syntax-highlighting-code-block/blob/99510a21e59603edaae435620464cf11eaf3444b/src/index.js#L136-L137

Perhaps it is due to this apparent race condition I commented about on the subsequent lines:

https://github.com/westonruter/syntax-highlighting-code-block/blob/99510a21e59603edaae435620464cf11eaf3444b/src/index.js#L139-L143

In any case, capturing the original `save` the first time we see it seems to fix this:

```diff
--- a/src/index.js
+++ b/src/index.js
@@ -26,6 +26,8 @@ import * as BlockEditor from '@wordpress/block-editor';
  */
 import languagesNames from './language-names';
 
+let save;
+
 /**
  * Extend code block with syntax highlighting.
  *
@@ -37,6 +39,14 @@ const extendCodeBlockWithSyntaxHighlighting = (settings) => {
 		return settings;
 	}
 
+	if (settings.save) {
+		save = settings.save;
+	}
+
+	if (!save) {
+		return settings;
+	}
+
 	const useLightBlockWrapper =
 		settings.supports &&
 		settings.supports.lightBlockWrapper &&
@@ -269,6 +279,8 @@ const extendCodeBlockWithSyntaxHighlighting = (settings) => {
 			);
 		},
 
+		save,
+
 		deprecated: [
 			...(settings.deprecated || []),
 			{
```

But this hardly seems like the right solution.

@youknowriad Do you know what's going on here?

Reply:
I think that filter might run for deprecations too. So it's possible that one of them don't have save.